### DISCUSSION TITLE
Route Internationalization

### DISCUSSION THREAD

**Discussion by TomJD** [amannn/next-intl#866] [Q&A] [answered]
Hello!

I am building an app with `next-intl`, and I have some routes that as part of the folder structure are like in a certain language ie

`src/app/[locale]/(misc)/about-us`

and I want to achieve that they become translated at build time. If I change the locale from `en `to `fr`, the page route will be updated and instead of `about-us` I would get `a-propos-nous`. However, I noticed that the build is prerendering all the locales to the same route. 

So the output is

```
/[locale]/about-us
|-/en/about-us
|-/fr/about-us
|-/de/about-us
```

and so on. What could I have done wrong?

**[locale]/layout.tsx**
```
export function generateStaticParams() {
  return locales.map((locale) => ({ locale }));
}

export default async function RootLayout({
  children,
  params: { locale },
}: Props) {
  unstable_setRequestLocale(locale);

  return (
    <html lang={locale} className={inter.variable}>
      <body className="flex min-h-screen flex-col justify-start bg-neutral-50">
        <Suspense>
          <main>{children}</main>
        </Suspense>
        <Footer />
      </body>
    </html>
  );
}
```

**[locale]/page.tsx**

```
export default function Page(props: TProps) {
  unstable_setRequestLocale(props.params.locale);
  return <Hero />;
}
```

**[locale]/(misc)/about-us.tsx**

```
export default function AboutUs({ params: { locale } }: any) {
  unstable_setRequestLocale(locale);
}
```

**Comment by amannn** [comment]
Is there a bug in your app or are you concerned about the build output? Note that the middleware is responsible for creating the necessary rewrites when localized pathnames are used.

This recent discussion is closely related: https://github.com/amannn/next-intl/issues/846

**Reply by TomJD** [reply]
There is no bug per se, everything seems to be working as expected. My concern is the build output. Is it normal to not output the routes which are actually equivalent to the rewrites?

**Reply by amannn** [reply, marked as answer]
It’s just the way Next.js works, the linked discussion has some more details on this.
